firebase-ai/src/main/kotlin/com/google/firebase/ai/type/ThinkingConfig.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.ai.type
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/** Configuration parameters for thinking features. */
+public class ThinkingConfig(public val thinkingBudget: Int?) {
```

Comment:
Should it have a public constructor with a nullable `thinkingBudget` param? AFAIK it can only be created by the dev, and not returned, thus we can be more prescriptive with the public API

Reply:
Good point! I have made it a private constructor